### PR TITLE
Add ability to search fares by award points

### PR DESCRIPTION
**Pull request by Baril27**

### CODE REVIEW

**Comment by Baril27** [comment]
This is an initial attempt to solve issue #11. I still need to workout a clean way to show fare units ($ or SWA).

**Comment by jwiebalk** [comment]
In #14 I also added `Fare Type: ${fareType},` to the `Settings` box on the dashboard, might be helpful to throw that in also.

**Comment by ezekg** [comment]
Good start. I think we should:

1. Make sure to upcase the fare type, so we're not requiring users to type `--fare-type POINTS` (unless of course, SWA doesn't care and lowercase submissions works).
2. Not a fan of killing the $. Maybe we should initialize a constant like `CURRENCY_SYMBOL` and depending on the current fare type, set it to either an empty string or $, and interpolate that where the old $ symbols used to be.

@jwiebalk does your PR have anything different that you think should be added?

**Comment by dprothero** [comment]
What about a `format_price()` function? That way it can prepend `$` for `DOLLARS` and perhaps append ` pts` for `POINTS`?

**Comment by Baril27** [comment]
@dprothero Like the `format_price()` solution. Going to add that in.

**Comment by Baril27** [comment]
Sorry guys didn't get back to this as fast as I wanted to.
- Added the units helper for `pts` and `$`.
- Upcased the fare type.

Let me know what you guys see.

@jwiebalk Think I have the faretype in the settings dashboard https://github.com/ezekg/swa-dashboard/pull/13/files#diff-168726dbe96b3ce427e7fedce31bb0bcR503. Can you confirm.

File: index.js [hunk 1]
```diff
@@ -50,6 +51,10 @@ process.argv.forEach((arg, i, argv) => {
     case "--return-date":
       returnDateString = argv[i + 1]
       break
+    case "--fare-type":
+      fareType = argv[i + 1]
+      fareType = fareType.toUpperCase()
```
Comment (ezekg) [inline index.js]: Should shorten this to `argv[i + 1].toUpperCase()`.

File: index.js [hunk 2]
```diff
@@ -335,6 +340,22 @@ const sendTextMessage = (message) => {
...
+ *
+ * @param {Int} price
+ *
+ * @return {Str}
+ */
+const formatPrice = (price) => {
+  if (fareType == 'POINTS') {
+    var price = `${price} pts`
+  } else {
+    var price = `\$${price}`
```
Comment (ezekg) [inline index.js]: No need to redefine with `var` here since `price` is already defined via the param.

File: index.js [hunk 3]
```diff
@@ -358,14 +379,24 @@ const fetch = () => {
...
     .find("#faresReturn .product_price")
     .then((priceMarkup) => {
-      const matches = priceMarkup.toString().match(/\$.*?(\d+)/)
-      const price = parseInt(matches[1])
+      if (fareType == 'POINTS') {
+        const matches = priceMarkup.text().split(',').join('')
+        var price = parseInt(matches)
+      } else {
+        const matches = priceMarkup.toString().match(/\$.*?(\d+)/)
+        var price = parseInt(matches[1])
```
Comment (ezekg) [inline index.js]: Should use `const` instead of `var` since the value never changes. May also be easier to extract this logic out into a function so we can avoid duplication.

**Comment by jwiebalk** [comment]
@Baril27 ahh yep must have missed that. 👍

File: index.js [hunk 4]
```diff
@@ -469,10 +500,11 @@ dashboard.settings([
   `Destination airport: ${destinationAirport}`,
   `Outbound date: ${outboundDateString}`,
   `Return date: ${returnDateString}`,
+  `Fare Type: ${fareType}`,
```
Comment (ezekg) [inline index.js]: Let's change "Fare Type" to "Fare type" (lowercase "type") to follow the case conventions of the other options.

**Comment by ezekg** [comment]
@Baril27 I added comments to https://github.com/ezekg/swa-dashboard/pull/13/files#diff-168726dbe96b3ce427e7fedce31bb0bcR503 – most a minor style changes. Looking good. 👍

File: index.js [hunk 5]
```diff
@@ -335,6 +340,22 @@ const sendTextMessage = (message) => {
...
 
 /**
+ * Format fare type price
+ *
+ * @param {Int} price
+ *
+ * @return {Str}
+ */
+const formatPrice = (price) => {
+  if (fareType == 'POINTS') {
```
Comment (ezekg) [inline index.js]: Use strict equality comparison: `===`.

File: index.js [hunk 6]
```diff
@@ -347,11 +347,27 @@ const sendTextMessage = (message) => {
...
+
+/**
+ * Parse and return pricing from HTML markup
+ *
+ * @param {Str} priceMarkup
+ *
+ * @return {Int}
+ */
+const parsePriceMarkup = (priceMarkup) => {
+  if (fareType == 'POINTS') {
```
Comment (ezekg) [inline index.js]: Update this to use strict equality.

**Comment by ezekg** [comment]
Looking good! One last minor update and I can get this merged ASAP if you're ready.

File: index.js (hunk 6 above)
Comment (Baril27) [inline index.js]: Ahh my bad missed that when I moved it. Updating now.

**Comment by Baril27** [comment]
@ezekg Added that last style change. Ready whenever you are.

**Comment by ezekg** [comment]
Great job. 👏
